### PR TITLE
verify the file name length not to exceed 255 characters + verify path d...

### PR DESCRIPTION
**Pull request by DeepDiver1975**
...uring rename

refs #14629

### CODE REVIEW

File: lib/private/files/view.php [hunk 1]
```diff
@@ -585,6 +586,8 @@ public function rename($path1, $path2) {
 				);
 			}
 			if ($run) {
+				$this->verifyPath(dirname($path1), basename($path1));
```
Comment (nickvergessen) [inline lib/private/files/view.php]: Does it make sense to verify the `path1`? We are just moving away from that. Shouldn't matter, if it doesn't match the criteria anymore?

**Comment by karlitschek** [comment]
makes sense :+1:

File: lib/private/files/view.php (hunk 1 above)
Comment (DeepDiver1975) [inline lib/private/files/view.php]: true story - thx

**Comment by nickvergessen** [comment]
Didn't test, but code looks good :+1:

**Comment by DeepDiver1975** [comment]
feedback from client devs would be great @dragotin @danimo @guruz @ogoffart THX

**Comment by guruz** [comment]
I think @danimo and @dragotin have quite strong opinions on this.

Note that Windows actually does support longer filenames if you use the API functions the right way
http://stackoverflow.com/a/265782/2941
(Which the owncloud client currently doesn't)

**Comment by DeepDiver1975** [comment]
> Note that Windows actually does support longer filenames if you use the API functions the right way

Well - we have this hard limitation on the server - further ref: http://serverfault.com/questions/9546/filename-length-limits-on-linux

There is not much we can do besides give back a proper exception to you.

**Comment by dragotin** [comment]
@guruz note this is about **file** names, not the entire path length.

**Pull request merged**